File: app/src/main/java/com/example/josycom/flowoverstack/viewmodel/CustomQuestionViewModelFactory.java
package com.example.josycom.flowoverstack.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class CustomQuestionViewModelFactory implements ViewModelProvider.Factory {

    private int page;
    private int pageSize;
    private String order;
    private String sortCondition;
    private String site;
    private String filter;

    public CustomQuestionViewModelFactory(int page, int pageSize, String order, String sortCondition, String site, String filter){
        this.page = page;
        this.pageSize = pageSize;
        this.order = order;
        this.sortCondition = sortCondition;
        this.site = site;
        this.filter = filter;
    }
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new QuestionViewModel(page, pageSize, order, sortCondition, site, filter);
    }
}

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/viewmodel/CustomQuestionViewModelFactory.kt
package com.josycom.mayorjay.flowoverstack.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.josycom.mayorjay.flowoverstack.data.repository.QuestionRepository
import javax.inject.Inject

class CustomQuestionViewModelFactory @Inject constructor(private val questionRepository: QuestionRepository) : ViewModelProvider.Factory {

    private var page = 0
    private var pageSize = 0
    private var order: String = ""
    private var sortCondition: String = ""
    private var site: String = ""
    private var tagged: String = ""
    private var filter: String = ""
    private var siteKey: String = ""

    fun setInputs(page: Int, pageSize: Int, order: String, sortCondition: String, site: String, tagged: String, filter: String, siteKey: String) {
        this.page = page
        this.pageSize = pageSize
        this.order = order
        this.sortCondition = sortCondition
        this.site = site
        this.tagged = tagged
        this.filter = filter
        this.siteKey = siteKey
    }

    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        when {
            modelClass.isAssignableFrom(QuestionViewModel::class.java) -> return QuestionViewModel(
                questionRepository, page, pageSize,
                order, sortCondition, site, tagged, filter, siteKey
            ) as T
            else -> throw IllegalArgumentException("Wrong Class Provided $modelClass")
        }
    }
}